src/addrdb.cpp

```diff
@@ -196,6 +196,11 @@ std::optional<bilingual_str> LoadAddrman(const std::vector<bool>& asmap, const A
         addrman = std::make_unique<AddrMan>(asmap, /* deterministic */ false, /* consistency_check_ratio */ check_addrman);
         LogPrintf("Creating peers.dat because the file was not found (%s)\n", fs::quoted(fs::PathToString(path_addr)));
         DumpPeerAddresses(args, *addrman);
+    } catch (const InvalidAddrManVersionError&) {
+        // Addrman can be in an inconsistent state after failure, reset it
+        addrman = std::make_unique<AddrMan>(asmap, /* deterministic */ false, /* consistency_check_ratio */ check_addrman);
+        LogPrintf("Creating new peers.dat because the file version was not compatible (%s)\n", fs::quoted(fs::PathToString(path_addr)));
```

Comment:
I wonder if the old file should be backed up first.

Reply:
I am not sure if that would be useful. Anyone who cares about which peers they connect to is probably managing a conf file filled with add peer directives.

Reply:
Agree with _luke-jr_

Reply:
I just saw your issue on knots.

I will make a backup at `peers.dat.bak`.

Reply:
The behavior has changed to back up to .bak file. Please let me know what you think of the commit before I squash it.